170 fix issues in vehicle structure (#171)
asam-ev/OpenMATERIAL: content/geometry/object-vehicle/vehicle-structure.adoc

| Perpendicular to the x-axis, pointing vertically upwards
|===

== Grp_Exterior_Dynamic

This group contains all dynamic parts of the vehicle's exterior. 

Dynamic parts are geometric structures whose position and orientation relative to the vehicle's origin may change throughout a simulation.
They may also change their state during the simulation, which is why lights are considered dynamic.

> Typically, this group do not move in the simulation and therefore do not have their own transforms.

Reviewer: ```suggestion
Typically, this group does not move in the simulation and therefore does not have its own transforms.
```

| Geometric center of the virtual hinge axis

| *x-axis*

> | Perpendicular to the z-axis, pointing towards the door

Author: pointing along the closed door sounds good

> | Concentric and coaxial to the virtual hinge axis

Reviewer: In https://github.com/asam-ev/OpenMATERIAL/pull/146/files, we decided that the direction of the z-axis is crucial for the correct direction of the y-axis. This is my proposal for the new z-axis definition:

"Concentric and coaxial to the virtual hinge axis, pointing in the direction that enables the door to move within the xy-plane"

Author: [...] pointing in the direction that enables the door to open with a positive rotation around the z-axis.

Reviewer: Thanks, I will integrate the feedback today

Reviewer: Updated definition of z-axis across file

> | Center of the plate

Author: Done

|===

== Grp_Light_Warning_<warning_idx>

This group contains all parts of the vehicle's warning light.
Warning lights can include various emergency lights, hazard lights, and more.

`<warning_idx>` denotes the index of warning lights. The index entries
are sorted from right to left in positive y-direction, and from front to rear, starting with 0.

NOTE: Add a figure on an exemplary rotating warning light on the roof the vehicle.

.<Table title>
[%header, cols="20, 80"]
|===

2+^| <Header>

| *Origin*
| Center of the light element

| *x-axis*
| Pointing towards the main light emission, or forwards for rotating lights

| *y-axis*
| Completes the right-handed coordinate system

| *z-axis*
| Perpendicular to the x-axis, pointing vertically upwards
|===

== Grp_Mirror_Blindspot_Joint_Left_<blindspot_mirror_joint_idx>

> This group contains all parts of the main moving part of a blindspot mirror, which adjusts when moved.

Reviewer: I am not sure, whether I understand the sentence correctly. Does the "main moving part" refer to the complete component consisting of the plastic/metal structure that holds the mirror as well as the mirror glass itself? Or are we only talking about the joint, which connects the plastic/metal structure with the vehicle?

My proposals

- A: "This group contains all parts of the movable structure that holds the blindspot mirror. The blindspot mirror automatically adjusts when the blindspot joint is moved."
- B. "This group contains all parts that connect the blindspot mirror mounting to the vehicle. The blindspot mirror automatically adjusts when the angle of the blindspot joint changes."

Author: Proposal A is correct. It contains all movable parts including the plastic or metal housing and the mirror glass itself (as a child node)

Reviewer: Integrated feedback

> It is a child of the corresponding mirror mounting.

Author: Yes that sounds good

> It is a child of the corresponding mirror mounting.

Reviewer: Updated descriptions across file

|===

== Grp_Mirror_Blindspot_Mounting_<blindspot_mirror_mounting_idx>

This group contains all parts of the vehicle's blindspot mirror mounting.

> It shall be a child of `<Grp_Exterior_Dynamic>` if mounted directly to the vehicle body, or a child of `<Grp_Door_Left>` or `<Grp_Door_Right>` if mounted to the door.

Reviewer: The modal verb shall is used inconsistently. For example, section Grp_Mirror_Blindspot_Joint_Left uses the following phrase in the same contex:

"It is a child of the corresponding mirror mounting."

For consistency and readability reasons, we should use consistent phrases. We can either remove the shall or, if we we want to emphasize that there are conditions that require different handling, rephrase to:

"If mounted to the vehicle body directly, it is a child node of the group `<Grp_Exterior_Dynamic>`. If mounted to the door it is a child node of `<Grp_Door_Left>` or `<Grp_Door_Right>`."

Author: Yes, let's remove the shall here.

Reviewer: Removed "shall" in this context across file

2+^| <Header>

| *Origin*
| Base of the mirror mounting

| *x-axis*
| Collinear with the vehicle longitudinal axis, pointing forwards

| *y-axis*
| Completes the right-handed coordinate system

| *z-axis*
| Perpendicular to the x-axis, pointing vertically upwards
|===

== Grp_Mirror_Side_View_Left_<side_mirror_view_left_idx>

This group is an empty node that represents the view direction of the mirror glass on a side mirror on the left side of the vehicle. 
It is a child of the corresponding mirror joint.

> The `<side_mirror_view_left_idx>` index numbers each mirrors on the left side, counting from right to left in positive y-direction, and from the front to the rear, starting with 0.

Reviewer: ```suggestion
The `<side_mirror_view_left_idx>` index numbers each mirror on the left side, counting from right to left in positive y-direction, and from the front to the rear, starting with 0.
```

2+^| <Header>

| *Origin*
| Center of the mirror glass surface

| *x-axis*
| Pointing outwards from the mirror glass, aligned with the surface normal

| *y-axis*
| Completes the right-handed coordinate system

| *z-axis*
| Perpendicular to the x-axis, pointing vertically upwards along the face of the mirror glass
|===

== Grp_Mirror_Side_View_Right_<side_mirror_view_right_idx>

This group is an empty node that represents the view direction of the mirror glass on a side mirror on the right side of the vehicle. 
It is a child of the corresponding mirror joint.

> The `<side_mirror_view_right_idx>` index numbers each mirrors on the right side, counting from right to left in positive y-direction, and from the front to the rear, starting with 0.

Reviewer: ```suggestion
The `<side_mirror_view_right_idx>` index numbers each mirror on the right side, counting from right to left in positive y-direction, and from the front to the rear, starting with 0.
```

| Perpendicular to the x-axis, pointing upwards in sensor coordinates
|===

== Grp_Wheel_<axle_idx>_<wheel_idx>

This group contains all geometries of a single wheel assembly, which may consist of the tire, rim, brake caliper, and so on.

`<axle_idx>` denotes the index of the axle to which the wheel is mounted, counting from the front to the rear, starting at 0.
`<wheel_idx>` denotes the index of the wheel on the specified axle, counting from right to left in the positive y-direction, starting at 0.
For example, the wheel on the front left of a standard vehicle would be labeled `Grp_Wheel_0_1`.

Wheel steering is represented by rotation around the z-axis.
Wheel camber is defined by a rotation around the x-axis.
Suspension deflection is represented by translation along the z-axis.

> Zero rotation and and translation around all axles are defined under neutral load conditions.

Reviewer: ```suggestion
Zero rotation and translation around all axles are defined under neutral load conditions.
```

|===

== Grp_Exterior_Static

This group contains all static parts of the vehicle's exterior.
Static elements are geometric structures that have a fixed position and
orientation relative to the vehicle's origin throughout the simulation.

> In contrast to lights, which change their state depending on whether they are

Reviewer: @ClemensLinnhoff: I rephrased this statement to enhance readability. Please check whether it is correct that lights are treated as dynamic elements because they can change state by switching on/off. Or is the main reason that the driver can adjust the direction of the light emission?

Author: Your phrasing is correct, thanks.

| Perpendicular to the x-axis, pointing vertically upwards
|===

== Grp_Interior_Dynamic

This group contains all dynamic parts of the vehicle's interior.  
Dynamic elements are geometric structures whose position and orientation relative to the vehicle's origin may change throughout the simulation.

> They may also change their state during the simulation; for example, lights are considered dynamic.

Reviewer: ```suggestion
They may also change their state during the simulation. Examples of dynamic elements are lights, which can be switched on and off.
```

| *z-axis*
| Perpendicular to the x-axis, pointing vertically upwards
|===

== Grp_Vehicle_Part

One or more optional vehicle parts may be added to the main vehicle structure.

> A vehicle part is a large component of the vehicle that moves based on other parts of the same vehicle.

Reviewer: ```suggestion
A vehicle part is a large component of a vehicle that moves in a slightly different direction than other parts, for example, the vehicle part follows an individual path during turns.
```

A vehicle may have multiple vehicle parts at the same hierarchy level or in a parent-child relationship.
This typically applies to articulated vehicles, such as a front loader with a hinged axis or an articulated bus with a hinge in the middle, and may also apply to construction vehicles.
When a vehicle part can be detached, like a trailer, it is treated as a separate object, not as a vehicle part.

The child nodes of a vehicle part may follow the same structure as the main vehicle. For example, if a vehicle part has lights, they may use the same structure and naming conventions as those on the main vehicle.

Standard passenger vehicles do not have separate parts.

.<Table title>
[%header, cols="20, 80"]
|===

2+^| <Header>

| *Origin*
| Center of the joint to the main vehicle or the parent.

| *x-axis*

> | Collinear with the parts longitudinal axis, pointing forwards

Reviewer: ```suggestion
| Collinear with the vehicle part's longitudinal axis, pointing forwards
```

2+^| <Header>

| *Origin*
| Origin of the vehicle

| *x-axis*
| Collinear with the vehicle's longitudinal axis, pointing forwards

| *y-axis*
| Completes the right-handed coordinate system

| *z-axis*
| Perpendicular to the x-axis, pointing vertically upwards
|===

> == Grp_Vehicle_Part

Author: I would propose to leave it in, because then people can already gather experience in that direction and bring that into the project for the next OpenMATERIAL version.
But I don't have a strong opinion about it.

> == Grp_Vehicle_Part

Author: You need to approve this PR, otherwise I cannot merge, because your change request is still open.

> == Grp_Vehicle_Part

Author: Never mind, I could dismiss your change request. Now I can merge.
